pkgs/development/cuda-modules/generic-builders/multiplex.nix

```diff
@@ -52,12 +52,27 @@ let
     lib.versionAtLeast cudaMajorMinorVersion package.minCudaVersion
     && lib.versionAtLeast package.maxCudaVersion cudaMajorMinorVersion;
 
-  # Releases for our platform and CUDA version.
+  # Releases for our CUDA version.
+  # Collapses the releases into a single list of packages, filtering out packages for unsupported CUDA versions and
+  # augmenting those supporting the current CUDA version with the redistArch to which the package belongs.
   # See ../modules/${pname}/releases/releases.nix
-  # allPackages :: List Package
-  allPackages = lib.filter satisfiesCudaVersion (
-    evaluatedModules.config.${pname}.releases.${redistArch} or [ ]
-  );
+  # allPackages :: List (Package + { redistArch: String })
+  allPackages =
+    let
+      redistArchs = lib.attrNames evaluatedModules.config.${pname}.releases;
+    in
+    lib.concatMap (
+      redistArch:
+      let
+        packages = evaluatedModules.config.${pname}.releases.${redistArch};
+      in
+      lib.concatMap (
+        package: lib.optionals (satisfiesCudaVersion package) [ ({ inherit redistArch; } // package) ]
+      ) packages
+    ) redistArchs;
+
+  packageOlder = p1: p2: lib.versionOlder p1.version p2.version;
+  packageSupportedPlatform = p: p.redistArch == redistArch;
```

Comment:
Or you keep the previous version but rename it to `allPackages'`, and add `allPackages = allPackages' ++ [(head ...)]` where `...` is flattened `releases`. The previous `filter` was more readable than 2x concatMap

Reply:
Re-introduced `allReleases` and introduced `allReleases'` to make it more clear what's happening.

Reply:
By "previous" I meant the one you introduced in the previous PR, where you only keep the matching platform's packages. That part looked fairly simple and comprehensible, we could keep it and append just one potentially-unsupported package at the end

Reply:
Your call though, I can also look in more detail at the current version?

Reply:
I'd say look more closely at the current version since it makes sure that the incompatible package used at least matches the version.

Reply:
> match the version

Good point. FYI my plan for the fallback is to `runCommand pname { ... } "false"`, but it might be easier to do after the evalModules schema update